### PR TITLE
fix: handle invalid json / string body

### PR DESCRIPTION
**Pull request by s-ashwinkumar**
Asana: https://app.asana.com/0/1198085188814749/1197421727948877

Changes are only in 
src/main/java/com/smartcar/sdk/SmartcarException.java
src/test/java/com/smartcar/sdk/SmartcarExceptionTest.java

### CODE REVIEW

File: src/main/java/com/smartcar/sdk/SmartcarException.java [hunk 1]
```diff
@@ -84,7 +85,16 @@ public String getRequestId() {
   }
 
   public static SmartcarException Factory(final Response response) throws IOException {
-    JsonObject body = gson.fromJson(response.body().string(), JsonObject.class);
+    JsonObject body;
```
Comment (s-ashwinkumar) [inline src/main/java/com/smartcar/sdk/SmartcarException.java:88]: Adding comment so this pops up on PR main page.

File: src/main/java/com/smartcar/sdk/SmartcarException.java [hunk 2]
```diff
@@ -84,7 +85,16 @@ public String getRequestId() {
...
+    try {
+      body = gson.fromJson(response.body().string(), JsonObject.class);
+    }catch (com.google.gson.JsonParseException exception) {
```
Comment (sankethkatta) [inline src/main/java/com/smartcar/sdk/SmartcarException.java:91]: nit: spacing
```suggestion
    } catch (com.google.gson.JsonParseException exception) {
```

File: src/main/java/com/smartcar/sdk/SmartcarException.java [hunk 3]
```diff
@@ -84,7 +85,16 @@ public String getRequestId() {
...
+      // In case the body is a string. Ex. gateway timeout where LB sends a non json body
+      String errorBody = Json.createObjectBuilder()
+              .add("message", response.body().string())
+              .build().toString();
+      body = gson.fromJson(errorBody, JsonObject.class);
+    }
```
Comment (sankethkatta) [inline src/main/java/com/smartcar/sdk/SmartcarException.java:97]: add another generic `catch` for any other error that could be thrown and raise it as a `SmartcarException`

**Review by sankethkatta — CHANGES_REQUESTED** [review]
This solves the issue at hand.

**Review by sankethkatta — APPROVED** [review]

**Pull request merged**